File: services/Espaces/src/main/java/com/capfest/espaces/Entity/Espaces.java
package com.capfest.espaces.Entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Espaces {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String nom;
    private String description;
    private String adresse;
    private Double latitude;
    private Double longitude;
    private Integer capacite;
    private Double tarif;
    private boolean disponibilite;

    @ElementCollection
    private List<String> photos;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categorie_id")
    @JsonIgnoreProperties("espaces")
    private CategorieEspace categorie;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY, mappedBy = "espace")
    @JsonManagedReference
    private List<Disponibilite> disponibilites;
}

File: src/model/Espaces.ts
import { Disponibilite } from './disponibilite';
import { CategorieEspace } from './CategorieEspace';

export interface Espaces {
  id?: number;
  nom: string;
  description: string;
  adresse: string;
  latitude: number;
  longitude: number;
  capacite: number;
  tarif: number;
  disponibilite: boolean;
  photos: string[];
  categorie?: CategorieEspace;
  disponibilites?: Disponibilite[];
}
